src/main/java/jsr223/perl/PerlScriptEngineFactory.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package jsr223.perl;
+
+import java.util.*;
+
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineFactory;
+
+import jsr223.perl.utils.PerlVersionGetter;
+import processbuilder.PerlSingletonPerlProcessBuilderFactory;
+import processbuilder.utils.PerlProcessBuilderUtilities;
+
+
+public class PerlScriptEngineFactory implements ScriptEngineFactory {
+
+    // Script engine parameters
+    private static final String NAME = "perl";
+
+    private static final String ENGINE = NAME;
+
+    private static final String ENGINE_VERSION = "0.1.0";
```

Comment:
Rather than having a static version value, you could use the same strategy as the bash engine to retrieve the perl interpreter version.